src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -632,12 +633,25 @@ ReactDOMComponent.Mixin = {
           this._currentElement.type
         );
       }
+      var isCustomComponentTag = isCustomComponent(this._tag, props);
+      if (__DEV__ && isCustomComponentTag && !didWarnShadyDOM && el.shadyRoot) {
+        var owner = this._currentElement._owner;
+        var name = owner.getName() || 'A component';
```

Comment:
Haven't seen a case where owner would be missing, but it seems like there are checks for owner in other places, added in latest commit to be safe.